### PR TITLE
Update Bio-Formats contributing pages

### PR DESCRIPTION
**Pull request by sbesson**
Fixes broken links in the contributing pages due the migration of the Bio-Formats merge/latest jobs out of ci-master into https://merge-ci.openmicroscopy.org/jenkins/.

Also simplifies Travis to only build the documentation including the latest Sphinx version.

### CODE REVIEW

File: contributing/ci-bio-formats.rst [hunk 1]
```diff
@@ -1,109 +1,45 @@
...
+    :header-rows: 1
 
-        -       * Builds the latest Bio-Formats artifacts using Maven
-                * :term:`BIOFORMATS-DEV-latest-maven`
+    -   * Job task
+        * Merge jobs
 
-        -       * Runs the daily Bio-Formats merge jobs
-                * :term:`BIOFORMATS-DEV-merge-daily`
+    -   * Merges the PRs and couple versions
```
Comment (mtbc) [inline contributing/ci-bio-formats.rst]: couples?

File: contributing/ci-bio-formats.rst [hunk 2]
```diff
@@ -1,109 +1,45 @@
...
-                #. Checks out :bf_scc_branch:`develop/merge/daily`
-                #. |buildBF|
-                #. Triggers :term:`BIOFORMATS-DEV-merge-matlab`
-
-        :jenkinsjob:`BIOFORMATS-DEV-merge-build-win`
+    :mergecijob:`BIOFORMATS-push`
 
-                This job builds the merge Bio-Formats artifacts using Ant
-                on Windows
+        This jobs merge all the PRs opened against the development branch of
```
Comment (mtbc) [inline contributing/ci-bio-formats.rst]: These? Or "This job merges"?

File: contributing/ci-bio-formats.rst [hunk 3]
```diff
@@ -1,109 +1,45 @@
...
+        https://github.com/ome/bio-formats-build and couple the component
```
Comment (mtbc) [inline contributing/ci-bio-formats.rst]: couples?

**Comment by sbesson** [comment]
@mtbc pushed

**Comment by mtbc** [comment]
That resolves my comments, ta.

**Comment by mtbc** [comment]
Non-blocking RFE: Elaborate to clarify the different functions of BIOFORMATS-build and BIOFORMATS-image. (At  the moment one does get a hint by reading the description of a different job.)

**Comment by mtbc** [comment]
Seems weird to me not to mention BIOFORMATS-build-docs at all on the docs jobs page.

**Comment by joshmoore** [comment]
> Seems weird to me not to mention BIOFORMATS-build-docs

Because we aren't happy with it?

For test-integration, I've been thinking about moving to a pipeline (perhaps `openmicroscopy/openmicroscopy:Jenkinsfile.test` or similar), perhaps @mtbc issue points to something similar for the B-F build:

 - `openmicroscopy/bioformats:Jenkinsfile` has a pipeline with stages:
    - build
    - docs
    - docker

File: contributing/editing-docs.rst [hunk 4]
```diff
@@ -87,7 +87,7 @@ Building/reviewing PRs via the CI
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Once a PR is open, you can build it for review using the
-:jenkinsjob:`BIOFORMATS-DEV-merge-docs` job on
+:mergecijob:`BIOFORMATS-build-docs` job on
 the Jenkins CI. Staging documentation is no longer deployed at a URL but you
 can download them as a zip for review with the correct styling by going to the
```
Comment (mtbc) [inline contributing/editing-docs.rst]: Maybe while here can clean up the is/are it/them tenses in these staging sentences. :smiley:

**Comment by mtbc** [comment]
Note also that the `editing-docs` page still has,

> with the next release - OMERO pyramid format

**Comment by mtbc** [comment]
> > Seems weird to me not to mention BIOFORMATS-build-docs

> Because we aren't happy with it?

We do mention it on the "Bio-Formats jobs" page, don't know if we then want a warning that it's expected to change. At least then an RFE to mention on the "Documentation jobs" page once we're happy with it; we mention OMERO-docs, etc. there and it's a reasonable place to go looking.

**Comment by sbesson** [comment]
Pushed a few commits addressing a few of the last comments including a reference to the CI Bio-Formats page from the CI Documentation page.

Note the initial scope of this job was to fix the contributing documentation linkchecker now failing following the CI jobs migration/decommission. Happy to take [commit suggestions](https://help.github.com/en/articles/reviewing-proposed-changes-in-a-pull-request), but I will probably stop adding to this PR before it becomes too big.

**Comment by joshmoore** [comment]
https://github.com/account/ssh might need an exception for a non-logged-in link checker:

```
    [exec] Warning, treated as error:
     [exec] /opt/hudson/workspace/CONTRIBUTING-merge-docs/src/contributing/source-code.rst.rst:87:broken link: https://github.com/account/ssh (404 Client Error: Not Found for url: https://github.com/account/ssh)
```

Unsure why this is showing up now.

**Comment by joshmoore** [comment]
Leaving up to you if you'd like to get that in here. With the other GH changes re: public -> authenticated, we might need an entire PR.

This PR LGTM. :+1:

**Comment by sbesson** [comment]
Pushed 2 commits to add the GitHub URL and clean up unused URLs that should be ignored by the linkchecker - see https://ci.openmicroscopy.org/job/CONTRIBUTING-merge-docs/1685/

**Comment by joshmoore** [comment]
That job lists "no changes" but I assume that's some standard weirdness. The changes look good. GFI.

**Pull request merged**